matRad_calcLateralParticleCutOff.m

```diff
@@ -172,19 +182,22 @@
                         IX   =  find(r_mid  >  Sigma2 * 0.85*LcutDefaultSigma2 ,1 ,'first');   
                         machine.data(energyIx).LatCutOff.CompFac = CF(LcutDefaultSigma2);
                       end  
-
-                case '1D'
-                      IX = find(dose_r  < (1-cutOffLevel) * DosePeakPos ,1 ,'first');   
-                      machine.data(energyIx).LatCutOff.CompFac = cutOffLevel^-1;
+    
                 case '2D'
                       cumArea = cumsum(2*pi.*r_mid.*dose_r.*dr);
-                      IX = find(cumArea > cumArea(end) - (1-cutOffLevel) * IntDosePeak,1, 'first');
+                      relativeThreshold = 0.5; %in [%]
+                      if abs((cumArea(end)./(idd(j)))-1)*100 > relativeThreshold && (cumArea(end) > relativeThreshold * idd(peakIx))
```

Comment:
I do not understand this safety check. Can you elaborate?

Reply:
This function was introduces to check the shell integration. cumArea(end) contains the integrated lateral dose which should be the same as the IDD(x). We could also check if the 2D area simply equals one - or we delete this line of code. The second condition in the if block is to only perform this check where the current IDD is  greater than 0.5% of the max IDD to avoid numerical issues in the tail -> sigma2 can get broad which is then not covered so good anymore by the shell integration steps.